### ISSUE TITLE
[ERROR] No plugin found for prefix 'jib' in the current project and in the plugin groups...

### ISSUE THREAD

**Issue by Zephery** [GoogleContainerTools/jib#522]
[ERROR] No plugin found for prefix 'jib' in the current project and in the plugin groups [org.apache.maven.plugins, org.codehaus.mojo] available from the repositories [local (/Users/zepherywen/.m2/repository), nexus (http://x.x.x.x:8081/nexus/content/groups/public/), central (https://repo.maven.apache.org/maven2)] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/NoPluginFoundForPrefixException

**Comment by coollog** [comment]
Hi @Zephery , Jib should be available in the Maven Central repository (https://repo.maven.apache.org/maven2/com/google/cloud/tools/jib-maven-plugin/). Could you share your `pom.xml` or a snippet of it to help diagnose this issue?

**Comment by coollog** [comment]
Closing unless you are still experiencing this issue.

**Comment by KTannenberg** [comment]
To anyone who encounters same issue. 

In order to use short name for plugin, you need to add new `pluginGroup` to your `settings.xml`:
```
<settings>
  ...
  <pluginGroups>
    ...
    <pluginGroup>com.google.cloud.tools</pluginGroup>
    ...
  </pluginGroups>
  ...
</settings>
```
Once this is done you can use `mvn clean compile jib:<jib-goal>` instead of `mvn clean compile com.google.cloud.tools:jib-maven-plugin:<jib-goal>`

More details about plugin groups: https://maven.apache.org/guides/introduction/introduction-to-plugin-prefix-mapping.html

**Comment by loosebazooka** [comment]
That's strange, if I understand correctly, plugin prefix mapping should only be necessary if youre not explicitly declaring the plugin in your pom? Is that the case here?

**Comment by KTannenberg** [comment]
@loosebazooka, that should be the case, at least it works this way for pitest plugin.

It might be the case, that I've tested it on multimodule project and only declared jib plugin in submodule that actually needs to be published and then launched maven from project's root.

I've rechecked now and both of these works as expected:
```
cd project && mvn jib:build --projects :module
```
```
cd project/module && mvn jib:build
```

Adding jib to parent pom as `pluginManagement` solves this issue, but they jib will attempt to package every module into container and fails as there are no `target/` for pom-only modules (this is fixed in #734)

**Comment by lgd8069465** [comment]
[INFO] ------------------------------------------------------------------------
[ERROR] No plugin found for prefix 'jib' in the current project and in the plugin groups [org.apache.maven.plugins, org.codehaus.mojo] available from the repositories [local (E:\Develop\Maven\local_repository), central (https://repo.maven.apache.org/maven2)] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/NoPluginFoundForPrefixException


change aliyun

[INFO] ------------------------------------------------------------------------
[ERROR] No plugin found for prefix 'jib' in the current project and in the plugin groups [org.apache.maven.plugins, org.codehaus.mojo] available from the repositories [local (C:\Users\Administrator\.m2\repository), central (http://maven.aliyun.com/nexus/content/groups/public), snapshots (http://maven.aliyun.com/nexus/content/groups/public), rdc-releases (https://repo.rdc.aliyun.com/repository/43168-release-Ho1L6B/), rdc-snapshots (https://repo.rdc.aliyun.com/repository/43168-snapshot-PR5W0P/)] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/NoPluginFoundForPrefixException

 mvn -v
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8
Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-18T02:33:14+08:00)
Maven home: E:\Develop\Maven\apache-maven-3.5.4\bin\..
Java version: 1.8.0_162, vendor: Oracle Corporation, runtime: C:\Environment\JDK8\jre
Default locale: zh_CN, platform encoding: UTF-8
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"

my pom.xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
	<build>
    <plugins>
	<plugin>
        <groupId>com.google.cloud.tools</groupId>
        <artifactId>jib-maven-plugin</artifactId>
        <version>1.0.0</version>
        <configuration>
		  <from>
           <image>openjdk:alpine</image>
          </from>
          <to>
            <image>hub.io/jib/basedata-jib-test</image>
			<tags>
              <tag>201902151702</tag>
              <tag>latest</tag>
            </tags>
          </to>
        </configuration>
		<executions>
          <execution>
           <phase>package</phase>
           <goals>
             <goal>build</goal>
           </goals>
          </execution>
        </executions>
      </plugin>
	  </plugins>
  </build>
	  
</project>

**Comment by briandealwis** [comment]
@lgd8069465 please open new issues instead of piggy-backing on existing issues.  Then reference related issues.

Your pom.xml isn't valid: Maven 3.5.4 complains that it's missing a modelVersion, groupId, artifactId, and version.  When I add those in, Maven is able to resolve `jib` via `mvn jib:build` though it complains since I have no classes etc. to package.

**Comment by lgd8069465** [comment]
@briandealwis     
my bad 
here is my pom.xml
```
<?xml version="1.0" encoding="utf-8"?>

<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">  
  <parent> 
    <groupId>com.lkc.zms</groupId>  
    <artifactId>zms</artifactId>  
    <version>0.0.1-SNAPSHOT</version> 
  </parent>  
  <modelVersion>4.0.0</modelVersion>  
  <artifactId>basedata</artifactId>  
  <version>0.0.1-SNAPSHOT</version>  
  <packaging>jar</packaging>   
  <build> 
    <plugins> 
      <plugin> 
        <groupId>com.google.cloud.tools</groupId>  
        <artifactId>jib-maven-plugin</artifactId>  
        <version>1.0.0</version>  
        <configuration> 
          <to> 
            <image>hub.io/jib/basedata-jib-test</image>  
            <tags> 
              <tag>201902180927</tag> 
            </tags> 
          </to> 
        </configuration> 
      </plugin> 
    </plugins> 
  </build> 
</project>

```

but it's still wrong
```
[INFO] ------------------------------------------------------------------------
[ERROR] No plugin found for prefix 'jib' in the current project and in the plugin groups [org.apache.maven.plugins, org.codehaus.mojo] available from the repositories [local (E:\Develop\Maven\local_repository), central (https://repo.maven.apache.org/maven2)] -> [Help 1]
org.apache.maven.plugin.prefix.NoPluginFoundForPrefixException: No plugin found for prefix 'jib' in the current project and in the plugin groups [org.apache.maven.plugins, org.codehaus.mojo] available from the repositories [local (E:\Develop\Maven\local_repository), central (https://repo.maven.apache.org/maven2)]
    at org.apache.maven.plugin.prefix.internal.DefaultPluginPrefixResolver.resolve (DefaultPluginPrefixResolver.java:95)
    at org.apache.maven.lifecycle.internal.MojoDescriptorCreator.findPluginForPrefix (MojoDescriptorCreator.java:266)
    at org.apache.maven.lifecycle.internal.MojoDescriptorCreator.getMojoDescriptor (MojoDescriptorCreator.java:220)
    at org.apache.maven.lifecycle.internal.DefaultLifecycleTaskSegmentCalculator.calculateTaskSegments (DefaultLifecycleTaskSegmentCalculator.java:104)
    at org.apache.maven.lifecycle.internal.DefaultLifecycleTaskSegmentCalculator.calculateTaskSegments (DefaultLifecycleTaskSegmentCalculator.java:83)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:89)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
[ERROR]
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/NoPluginFoundForPrefixException
```

i refer to https://github.com/mfriedenhagen/dummy-lifecycle-mapping-plugin

**Comment by chanseokoh** [comment]
Haven't tried, but I guess your `pom.xml` is still missing `<groupId>`.

**Comment by briandealwis** [comment]
@chanseokoh there's a `groupId` in the `parent`.

@lgd8069465 your example works for me when I comment out your parent (since I don't have access to it) and copy the `groupId`.  At this point, this seems to be a problem with your local Maven installation.   Please open a [new issue](
https://github.com/GoogleContainerTools/jib/issues/new) and include your full build log.

**Comment by briandealwis** [comment]
@lgd8069465 I apologize for locking this issue, and you were completely right to comment on this issue.

I just hit this issue in creating an example multi-module project for Skaffold that uses Jib, and now I understand @lgd8069465's example as well as @KTannenberg's fix: the problem is the `mvn jib:xxx` is being run from the parent which is not configuring the `jib-maven-plugin`, which is instead being configured in a child project.  So running `mvn jib:xxx` from the parent project will fail as jib hasn't been referenced.

So a fix is to reference `jib-maven-plugin` from the parent, such as through a `<pluginManagement>` block:
```xml
  <build>
    <pluginManagement>
      <plugins>
        <plugin>
          <groupId>com.google.cloud.tools</groupId>
          <artifactId>jib-maven-plugin</artifactId>
          <version>1.0.0</version>
        </plugin>
      </plugins>
    </pluginManagement>
  </build>
```

**Comment by lgd8069465** [comment]
mvn compile jib:build --Djib.allowInsecureRegistries=true -X

```
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:1.0.0:build (default-cli) on project common: Tried to check BLOB exists for hub.io/jib/common-jib-test with digest sha256:6c40cc604d8e4c121adcb6b0bfe8bb038815c350980090e74aa5a6423f8f82c0 but failed because: Did not receive Content-Length header | If this is a bug, please file an issue at https://github.com/GoogleContainerTools/jib/issues/new -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.google.cloud.tools:jib-maven-plugin:1.0.0:build (default-cli) on project common: Tried to check BLOB exists for hub.io/jib/common-jib-test with digest sha256:6c40cc604d8e4c121adcb6b0bfe8bb038815c350980090e74aa5a6423f8f82c0 but failed because: Did not receive Content-Length header | If this is a bug, please file an issue at https://github.com/GoogleContainerTools/jib/issues/new
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Tried to check BLOB exists for hub.io/jib/common-jib-test with digest sha256:6c40cc604d8e4c121adcb6b0bfe8bb038815c350980090e74aa5a6423f8f82c0 but failed because: Did not receive Content-Length header | If this is a bug, please file an issue at https://github.com/GoogleContainerTools/jib/issues/new
    at com.google.cloud.tools.jib.maven.BuildImageMojo.execute (BuildImageMojo.java:163)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at浜屾湀 21, 2019 11:00:59 涓婂崍 com.google.common.util.concurrent.AggregateFuture$RunningState handleException
涓ラ噸: Got more than one input Future failure. Logging failures after the first
java.util.concurrent.RejectedExecutionException: Task com.google.cloud.tools.jib.plugins.common.logging.AnsiLoggerWithFooter$$Lambda$63/72615125@79339b97 rejected from java.util.concurrent.ThreadPoolExecutor@421b2cc5[Terminated, pool size = 0, active threa ds = 0, queued tasks = 0, completed tasks = 1000]
        at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2063)
        at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:830)
        at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1379)
o       at java.util.concurrent.Executors$DelegatedExecutorService.execute(Executors.java:668)
        at com.google.cloud.tools.jib.plugins.common.logging.SingleThreadedExecutor.execute(SingleThreadedExecutor.java:62)
        at com.google.cloud.tools.jib.plugins.common.logging.AnsiLoggerWithFooter.log(AnsiLoggerWithFooter.java:103)
        at com.google.cloud.tools.jib.maven.MavenProjectProperties.lambda$makeEventHandlers$0(MavenProjectProperties.java:118)
rg      at com.google.cloud.tools.jib.event.Handler.handle(Handler.java:40)
        at com.google.cloud.tools.jib.event.DefaultEventDispatcher.lambda$dispatch$1(DefaultEventDispatcher.java:44)
        at com.google.common.collect.ImmutableList.forEach(ImmutableList.java:407)
        at com.google.cloud.tools.jib.event.DefaultEventDispatcher.dispatch(DefaultEventDispatcher.java:44)
.       aat com.google.cloud.tools.jib.registry.RegistryEndpointCaller.fallBackToHttp(RegistryEndpointCaller.java:188)
pache   at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.callWithAllowInsecureRegistryHandling(RegistryEndpointCaller.java.:161)
maven.l at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.call(RegistryEndpointCaller.java:142)
ife     at com.google.cloud.tools.jib.registry.RegistryClient.callRegistryEndpoint(RegistryClient.java:356)
cycle.inat com.google.cloud.tools.jib.registry.RegistryClient.checkBlob(RegistryClient.java:241)
erna    at com.google.cloud.tools.jib.builder.steps.PushBlobStep.call(PushBlobStep.java:110)
l.M     at com.google.cloud.tools.jib.builder.steps.PushBlobStep.call(PushBlobStep.java:39)
oj      at com.google.common.util.concurrent.CombinedFuture$CallableInterruptibleTask.runInterruptibly(CombinedFuture.java:181)
oEx     at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:57)
ecut    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
or      .at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
e       at java.lang.Thread.run(Thread.java:748)
xecute  Suppressed: java.util.concurrent.RejectedExecutionException: Task com.google.cloud.tools.jib.plugins.common.logging.AnsiLoggerWithFooter$$Lambda$63/72615125@64c7fd10 rejected from java.util.concurrent.ThreadPoolExecutor@421b2cc5[T erminated, pool size = 0,( active threads = 0, queued tasks = 0, completed tasks = 1000]
Moj             at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2063)
oE              at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:830)
xec             at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1379)
u               at java.util.concurrent.Executors$DelegatedExecutorService.execute(Executors.java:668)
to              at com.google.cloud.tools.jib.plugins.common.logging.SingleThreadedExecutor.execute(SingleThreadedExecutor.java:62)
r.j             at com.google.cloud.tools.jib.plugins.common.logging.AnsiLoggerWithFooter.log(AnsiLoggerWithFooter.java:103)
a               at com.google.cloud.tools.jib.maven.MavenProjectProperties.lambda$makeEventHandlers$1(MavenProjectProperties.java:121)
va:             at com.google.cloud.tools.jib.plugins.common.TimerEventHandler.accept(TimerEventHandler.java:79)
1               at com.google.cloud.tools.jib.plugins.common.TimerEventHandler.accept(TimerEventHandler.java:24)
4               at com.google.cloud.tools.jib.event.Handler.handle(Handler.java:40)
6)              at com.google.cloud.tools.jib.event.DefaultEventDispatcher.lambda$dispatch$1(DefaultEventDispatcher.java:44)

                at com.google.common.collect.ImmutableList.forEach(ImmutableList.java:407)
                at com.google.cloud.tools.jib.event.DefaultEventDispatcher.dispatch(DefaultEventDispatcher.java:44)
                at com.google.cloud.tools.jib.builder.TimerEventDispatcher.dispatchTimerEvent(TimerEventDispatcher.java:101)
at              at com.google.cloud.tools.jib.builder.TimerEventDispatcher.close(TimerEventDispatcher.java:97)
 o              at com.google.cloud.tools.jib.builder.steps.PushBlobStep.$closeResource(PushBlobStep.java:125)
rg              at com.google.cloud.tools.jib.builder.steps.PushBlobStep.call(PushBlobStep.java:125)
.a              ... 6 more
p       Suppressed: java.util.concurrent.RejectedExecutionException: Task com.google.cloud.tools.jib.plugins.common.logging.AnsiLoggerWacithFooter$$Lambda$93/824915171@1fad266 rejected from java.util.concurrent.ThreadPoolExecutor@421b2cc5[Terminated, pool size = 0,he active threads = 0, queued tasks = 0, completed tasks = 1000]
.m              at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2063)
a               at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:830)
ve              at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1379)
n.l             at java.util.concurrent.Executors$DelegatedExecutorService.execute(Executors.java:668)
i               at com.google.cloud.tools.jib.plugins.common.logging.SingleThreadedExecutor.execute(SingleThreadedExecutor.java:62)
fec             at com.google.cloud.tools.jib.plugins.common.logging.AnsiLoggerWithFooter.setFooter(AnsiLoggerWithFooter.java:133)
yc              at com.google.cloud.tools.jib.maven.MavenProjectProperties.lambda$makeEventHandlers$2(MavenProjectProperties.java:126)
le.             at com.google.cloud.tools.jib.event.progress.ProgressEventHandler.accept(ProgressEventHandler.java:92)
in              at com.google.cloud.tools.jib.event.progress.ProgressEventHandler.accept(ProgressEventHandler.java:30)
te              at com.google.cloud.tools.jib.event.Handler.handle(Handler.java:40)
rnal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
                at com.google.cloud.tools.jib.event.DefaultEventDispatcher.lambda$dispatch$1(DefaultEventDispatcher.java:44)
at              at com.google.common.collect.ImmutableList.forEach(ImmutableList.java:407)
 o              at com.google.cloud.tools.jib.event.DefaultEventDispatcher.dispatch(DefaultEventDispatcher.java:44)
rg              at com.google.cloud.tools.jib.builder.ProgressEventDispatcher.dispatchProgress(ProgressEventDispatcher.java:137)
.ap             at com.google.cloud.tools.jib.builder.ProgressEventDispatcher.close(ProgressEventDispatcher.java:124)
ach             at com.google.cloud.tools.jib.builder.steps.PushBlobStep.$closeResource(PushBlobStep.java:125)
e.              at com.google.cloud.tools.jib.builder.steps.PushBlobStep.call(PushBlobStep.java:125)
ma              ... 6 more
ven
.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultM浜屾湀 21, 2019 11:00:59 涓婂崍 com.google.common.util.concurrent.AggregateFuture$RunningState handleException
ave涓ラ噸: Got more than one input Future failure. Logging failures after the first
n.djava.util.concurrent.RejectedExecutionException: Task com.google.cloud.tools.jib.plugins.common.logging.AnsiLoggerWithFooter$$Laombda$63/72615125@5d282ae0 rejected from java.util.concurrent.ThreadPoolExecutor@421b2cc5[Terminated, pool Esize = 0, active threaxds = 0, queued tasks = 0, completed tasks = 1000]
ecu     at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2063)
te      at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:830)
        at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1379)
 (De    at java.util.concurrent.Executors$DelegatedExecutorService.execute(Executors.java:668)
f       at com.google.cloud.tools.jib.plugins.common.logging.SingleThreadedExecutor.execute(SingleThreadedExecutor.java:62)
        at com.google.cloud.tools.jib.plugins.common.logging.AnsiLoggerWithFooter.log(AnsiLoggerWithFooter.java:103)
        at com.google.cloud.tools.jib.maven.MavenProjectProperties.lambda$makeEventHandlers$0(MavenProjectProperties.java:118)
aul     at com.google.cloud.tools.jib.event.Handler.handle(Handler.java:40)
tM      at com.google.cloud.tools.jib.event.DefaultEventDispatcher.lambda$dispatch$1(DefaultEventDispatcher.java:44)
ave     at com.google.common.collect.ImmutableList.forEach(ImmutableList.java:407)
n.j     at com.google.cloud.tools.jib.event.DefaultEventDispatcher.dispatch(DefaultEventDispatcher.java:44)
        at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.fallBackToHttp(RegistryEndpointCaller.java:188)
a       at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.callWithAllowInsecureRegistryHandling(RegistryEndpointCaller.javav:161)
a       at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.call(RegistryEndpointCaller.java:142)
:30     at com.google.cloud.tools.jib.registry.RegistryClient.callRegistryEndpoint(RegistryClient.java:356)
5       at com.google.cloud.tools.jib.registry.RegistryClient.checkBlob(RegistryClient.java:241)
)       at com.google.cloud.tools.jib.builder.steps.PushBlobStep.call(PushBlobStep.java:110)
        at com.google.cloud.tools.jib.builder.steps.PushBlobStep.call(PushBlobStep.java:39)
        at com.google.common.util.concurrent.CombinedFuture$CallableInterruptibleTask.runInterruptibly(CombinedFuture.java:181)
        at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:57)

        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
         Suppressed: java.util.concurrent.RejectedExecutionException: Task com.google.cloud.tools.jib.plugins.common.logging.AnsiLoggerW ithFooter$$Lambda$63/72615125@f157cee rejected from java.util.concurrent.ThreadPoolExecutor@421b2cc5[ Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 1000]
at              at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2063)
 o              at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:830)
r               at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1379)
g.              at java.util.concurrent.Executors$DelegatedExecutorService.execute(Executors.java:668)
ap              at com.google.cloud.tools.jib.plugins.common.logging.SingleThreadedExecutor.execute(SingleThreadedExecutor.java:62)
ach             at com.google.cloud.tools.jib.plugins.common.logging.AnsiLoggerWithFooter.log(AnsiLoggerWithFooter.java:103)
e.m             at com.google.cloud.tools.jib.maven.MavenProjectProperties.lambda$makeEventHandlers$1(MavenProjectProperties.java:121)
av              at com.google.cloud.tools.jib.plugins.common.TimerEventHandler.accept(TimerEventHandler.java:79)
en              at com.google.cloud.tools.jib.plugins.common.TimerEventHandler.accept(TimerEventHandler.java:24)
.               at com.google.cloud.tools.jib.event.Handler.handle(Handler.java:40)
Def             at com.google.cloud.tools.jib.event.DefaultEventDispatcher.lambda$dispatch$1(DefaultEventDispatcher.java:44)
au              at com.google.common.collect.ImmutableList.forEach(ImmutableList.java:407)
lt              at com.google.cloud.tools.jib.event.DefaultEventDispatcher.dispatch(DefaultEventDispatcher.java:44)
Mav             at com.google.cloud.tools.jib.builder.TimerEventDispatcher.dispatchTimerEvent(TimerEventDispatcher.java:101)
en.             at com.google.cloud.tools.jib.builder.TimerEventDispatcher.close(TimerEventDispatcher.java:97)
doE             at com.google.cloud.tools.jib.builder.steps.PushBlobStep.$closeResource(PushBlobStep.java:125)
xe              at com.google.cloud.tools.jib.builder.steps.PushBlobStep.call(PushBlobStep.java:125)
cu              ... 6 more
te      Suppressed: java.util.concurrent.RejectedExecutionException: Task com.google.cloud.tools.jib.plugins.common.logging.AnsiLoggerW ithFooter$$Lambda$93/824915171@27a1995e rejected from java.util.concurrent.ThreadPoolExecutor@421b2cc5([Terminated, pool size = 0De, active threads = 0, queued tasks = 0, completed tasks = 1000]
f               at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2063)
aul             at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:830)
tM              at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1379)
av              at java.util.concurrent.Executors$DelegatedExecutorService.execute(Executors.java:668)
en              at com.google.cloud.tools.jib.plugins.common.logging.SingleThreadedExecutor.execute(SingleThreadedExecutor.java:62)
.j              at com.google.cloud.tools.jib.plugins.common.logging.AnsiLoggerWithFooter.setFooter(AnsiLoggerWithFooter.java:133)
a       v       at com.google.cloud.tools.jib.maven.MavenProjectProperties.lambda$makeEventHandlers$2(MavenProjectProperties.java:126)
a:1             at com.google.cloud.tools.jib.event.progress.ProgressEventHandler.accept(ProgressEventHandler.java:92)
9               at com.google.cloud.tools.jib.event.progress.ProgressEventHandler.accept(ProgressEventHandler.java:30)
                at com.google.cloud.tools.jib.event.Handler.handle(Handler.java:40)
                at com.google.cloud.tools.jib.event.DefaultEventDispatcher.lambda$dispatch$1(DefaultEventDispatcher.java:44)
                at com.google.common.collect.ImmutableList.forEach(ImmutableList.java:407)
                at com.google.cloud.tools.jib.event.DefaultEventDispatcher.dispatch(DefaultEventDispatcher.java:44)
                at com.google.cloud.tools.jib.builder.ProgressEventDispatcher.dispatchProgress(ProgressEventDispatcher.java:137)
                at com.google.cloud.tools.jib.builder.ProgressEventDispatcher.close(ProgressEventDispatcher.java:124)
                at com.google.cloud.tools.jib.builder.steps.PushBlobStep.$closeResource(PushBlobStep.java:125)
                at com.google.cloud.tools.jib.builder.steps.PushBlobStep.call(PushBlobStep.java:125)
                ... 6 more
2
)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: com.google.cloud.tools.jib.registry.RegistryErrorException: Tried to check BLOB exists for hub.io/jib/common-jib-test with digest sha256:6c40cc604d8e4c121adcb6b0bfe8bb038815c350980090e74aa5a6423f8f82c0 but failed because: Did not receive Content-Length header | If this is a bug, please file an issue at https://github.com/GoogleContainerTools/jib/issues/new
    at com.google.cloud.tools.jib.registry.RegistryErrorExceptionBuilder.build (RegistryErrorExceptionBuilder.java:108)
    at com.google.cloud.tools.jib.registry.BlobChecker.handleResponse (BlobChecker.java:55)
    at com.google.cloud.tools.jib.registry.BlobChecker.handleResponse (BlobChecker.java:36)
    at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.call (RegistryEndpointCaller.java:234)
    at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.fallBackToHttp (RegistryEndpointCaller.java:191)
    at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.callWithAllowInsecureRegistryHandling (RegistryEndpointCaller.java:161)
    at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.call (RegistryEndpointCaller.java:142)
    at com.google.cloud.tools.jib.registry.RegistryClient.callRegistryEndpoint (RegistryClient.java:356)
    at com.google.cloud.tools.jib.registry.RegistryClient.checkBlob (RegistryClient.java:241)
    at com.google.cloud.tools.jib.builder.steps.PushBlobStep.call (PushBlobStep.java:110)
    at com.google.cloud.tools.jib.builder.steps.PushBlobStep.call (PushBlobStep.java:39)
    at com.google.common.util.concurrent.CombinedFuture$CallableInterruptibleTask.runInterruptibly (CombinedFuture.java:181)
    at com.google.common.util.concurrent.InterruptibleTask.run (InterruptibleTask.java:57)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:624)
    at java.lang.Thread.run (Thread.java:748)
[ERROR]
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :common
```

**Comment by lgd8069465** [comment]
@briandealwis @KTannenberg @chanseokoh
Thank you very much
It's worked

**Comment by fezho** [comment]
Also got this issue, my root cause is somebody commented the` <parent> </parent>` tag in one of the child module's pom.xml, after uncomment it, the build works.
